core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -597,6 +597,20 @@ public TableMetadata replaceProperties(Map<String, String> rawProperties) {
         .build();
   }
 
+  /**
+   * Prune the unused partition specs from the table metadata.
+   *
+   * @param specsToRemove the partition specs to be removed
+   * @return the new table metadata with the unused partition specs removed
+   */
+  TableMetadata pruneUnusedSpecs(List<PartitionSpec> specsToRemove) {
```

Comment:
I probably was not very clear, but I don't think that this should be a separate method on `TableMetadata`. We are changing this approach to use the builder directly instead.